nxontology_ml/model.py

```diff
@@ -0,0 +1,152 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from enum import Enum
+from itertools import islice
+from pathlib import Path
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from catboost import CatBoostClassifier, FeaturesData, Pool
```

Comment:
Curious as to whether catboost is more appropriate than xgboost or lightgbm for our data? Looks nice and simple and I have no clue which is best.

Reply:
This choice was arbitrary tbh (I have mostly used xgboost in past professional experiences and wanted to give a try to Catboost). Benchmarking them against each other is something we should do in the near future.